src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableHashSet_1.cs

```diff
@@ -1003,6 +1008,27 @@ private static ImmutableHashSet<T> Wrap(SortedInt32KeyNode<HashBucket> root, IEq
         }
 
         /// <summary>
+        /// Gets the <see cref="IEqualityComparer{HashBucket}"/> to use.
+        /// </summary>
+        /// <param name="valueComparer">The value comparer for T.</param>
+        /// <returns>The equality comparer to use.</returns>
+        private static IEqualityComparer<HashBucket> GetHashBucketEqualityComparer(IEqualityComparer<T> valueComparer)
+        {
+            if (!typeof(ValueType).IsAssignableFrom(typeof(T)))
```

Comment:
@AtsushiKan So the replacement you proposed turns out to be rather problematic. It fails because `Type.IsValueType` doesn't exist at least in some profiles. And in some profiles `Type.GetTypeInfo()` doesn't exist either. So I may revert this change and go back to the old-school `IsAssignableFrom` approach unless you can suggest how to get this to work.